drake/systems/plants/RigidBodySystem.cpp

```diff
@@ -463,41 +496,59 @@ RigidBodyDepthSensor::RigidBodyDepthSensor(
     parseScalarValue(range_node, "max", max_range);
   }
 
+  cacheRaycastEndpoints();
+}
+
+void RigidBodyDepthSensor::cacheRaycastEndpoints() {
   raycast_endpoints.resize(3, num_pixel_rows * num_pixel_cols);
   for (size_t i = 0; i < num_pixel_rows; i++) {
     double pitch =
         min_pitch +
         (num_pixel_rows > 1 ? static_cast<double>(i) / (num_pixel_rows - 1)
                             : 0.0) *
-            (max_pitch - min_pitch);
+        (max_pitch - min_pitch);
     for (size_t j = 0; j < num_pixel_cols; j++) {
       double yaw =
           min_yaw +
           (num_pixel_cols > 1 ? static_cast<double>(j) / (num_pixel_cols - 1)
                               : 0.0) *
-              (max_yaw - min_yaw);
+          (max_yaw - min_yaw);
       raycast_endpoints.col(num_pixel_cols * i + j) =
           max_range *
           Vector3d(
               cos(yaw) * cos(pitch), sin(yaw),
               -cos(yaw) *
-                  sin(pitch));  // rolled out from roty(pitch)*rotz(yaw)*[1;0;0]
+              sin(pitch));  // rolled out from roty(pitch)*rotz(yaw)*[1;0;0]
     }
   }
 }
 
+
+
 Eigen::VectorXd RigidBodyDepthSensor::output(const double &t,
                                              const KinematicsCache<double> &rigid_body_state,
                                              const RigidBodySystem::InputVector<double>& u) const {
-  VectorXd distances(num_pixel_cols*num_pixel_rows);
+  const size_t num_distances = num_pixel_cols * num_pixel_rows;
+  VectorXd distances(num_distances);
+
   Vector3d origin = sys.getRigidBodyTree()->transformPoints(rigid_body_state,
                                                              Vector3d::Zero(),
-                                                             frame->frame_index, 0);
+                                                             frame->frame_index,0);
+
   auto raycast_endpoints_world = sys.getRigidBodyTree()->transformPoints(rigid_body_state,
                                                                           raycast_endpoints,
-                                                                          frame->frame_index, 0);
-  sys.getRigidBodyTree()->collisionRaycast(rigid_body_state, origin, raycast_endpoints, distances);
-  // maybe: could publish lcm here (for debugging), since I can easily infer the 3D location in space
+                                                                          frame->frame_index,0);
+
+  sys.getRigidBodyTree()->collisionRaycast(rigid_body_state, origin, raycast_endpoints_world, distances);
```

Comment:
No worries.  It was easy enough to track down once I had the visualizer working.  Thanks @patmarion for the help there.

I think this was at least partly at fault for the bad raycast accuracy in #1712 but some rippling effects are still showing up in the visualizer.  I was able to back off the tolerance in lidarTest by a factor of 5.